In [1]:
import pandas as pd
import numpy as np

In [2]:
df2019=pd.read_csv('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/interim/d2019.csv' , sep =";")
df2019

Día     Ene     Feb     Mar     Abr     May     Jun     Jul     Ago  \
0     1     NaN  657.81  651.79  678.53     NaN     NaN  679.15  700.82   
1     2  694.77     NaN     NaN  672.56  678.71     NaN  677.62  704.88   
2     3  697.09     NaN     NaN  670.05  683.31  709.80  681.30     NaN   
3     4  697.64  655.39  656.57  667.55     NaN  704.59  680.48     NaN   
4     5     NaN  653.80  659.07  666.32     NaN  696.78  677.64  711.13   
5     6     NaN  652.78  657.49     NaN  678.68  693.25     NaN  721.70   
6     7  688.42  653.90  658.40     NaN  681.09  692.21     NaN  716.35   
7     8  679.69  654.44  662.19  664.17  683.49     NaN  683.12  718.55   
8     9  680.10     NaN     NaN  663.01  684.51     NaN  685.83  710.98   
9    10  678.09     NaN     NaN  663.15  686.43  692.28  688.88     NaN   
10   11  677.58  656.46  670.83  662.92     NaN  697.34  685.40     NaN   
11   12     NaN  660.82  669.37  664.19     NaN  693.60  683.38  712.04   
12   13     NaN  660.49  667.09     NaN  685.67  694.89     NaN  716.69   
13   14  675.15  661.60  665.68     NaN  693.04  696.25     NaN  708.37   
14   15  674.95  665.90  670.62  660.67  692.33     NaN  681.07     NaN   
15   16  673.76     NaN     NaN  662.86  692.62     NaN     NaN  710.14   
16   17  672.73     NaN     NaN  663.98  692.79  698.72  679.48     NaN   
17   18  671.37  663.98  669.39  660.48     NaN  699.52  681.69     NaN   
18   19     NaN  660.82  667.21     NaN     NaN  696.81  683.27  709.26   
19   20     NaN  658.75  664.78     NaN  696.61  693.99     NaN  711.94   
20   21  670.10  654.52  666.13     NaN     NaN  684.19     NaN  710.63   
21   22  672.42  654.33  665.12  662.27  697.23     NaN  685.81  709.52   
22   23  672.26     NaN     NaN  663.91  696.70     NaN  688.28  713.22   
23   24  672.41     NaN     NaN  668.19  698.03  683.41  691.57     NaN   
24   25  672.14  651.36  676.60  672.83     NaN  682.31  691.36     NaN   
25   26     NaN  649.22  679.08  676.21     NaN  678.32  694.16  716.40   
26   27     NaN  649.80  679.02     NaN  694.84  680.06     NaN  718.75   
27   28  667.04  649.92  683.73     NaN  696.46  679.86     NaN  721.55   
28   29  671.29     NaN  681.09  673.86  703.48     NaN  696.62  724.20   
29   30  669.60     NaN     NaN  677.67  708.20     NaN  697.22  720.65   
30   31  666.76     NaN     NaN     NaN  707.86     NaN  699.98     NaN   

       Sep     Oct     Nov     Dic  year  
0      NaN  728.21     NaN     NaN  2019  
1   720.20  729.38     NaN  812.13  2019  
2   724.72  727.80     NaN  804.29  2019  
3   725.69  719.53  735.05  802.44  2019  
4   723.30     NaN  736.66  796.75  2019  
5   717.52     NaN  745.85  784.25  2019  
6      NaN  715.53  743.65     NaN  2019  
7      NaN  717.05  740.23     NaN  2019  
8   711.22  722.28     NaN  779.04  2019  
9   713.72  724.27     NaN  771.29  2019  
10  714.77  720.61  746.23  778.14  2019  
11  714.39     NaN  756.08  772.93  2019  
12  711.29     NaN  783.72  768.69  2019  
13     NaN  712.98  794.51     NaN  2019  
14     NaN  712.04  801.83     NaN  2019  
15  707.07  715.48     NaN  766.76  2019  
16  709.81  718.69     NaN  762.83  2019  
17     NaN  713.23  782.82  756.95  2019  
18     NaN     NaN  773.81  752.30  2019  
19     NaN     NaN  782.71  752.60  2019  
20     NaN  709.71  797.17     NaN  2019  
21     NaN  723.52  795.14     NaN  2019  
22  715.24  726.24     NaN  749.83  2019  
23  722.17  726.15     NaN  754.27  2019  
24  721.92  723.43  795.94     NaN  2019  
25  727.08     NaN  791.16  751.04  2019  
26  726.17     NaN  793.49  746.66  2019  
27     NaN  725.87  806.30     NaN  2019  
28     NaN  724.37  828.25     NaN  2019  
29  725.68  726.34     NaN  744.62  2019  
30     NaN     NaN     NaN     NaN  2019

In [3]:
df2019.shape

(31, 14)

In [4]:
df2020=pd.read_csv('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/interim/d2020.txt', sep =" ")

In [5]:
df2020

Día     Ene     Feb     Mar     Abr     May     Jun     Jul     Ago  \
0     1     NaN     NaN     NaN  852.03     NaN  806.32  821.23     NaN   
1     2  748.74     NaN  818.32  863.81     NaN  796.46  816.29     NaN   
2     3  754.16  799.11  817.45  861.89     NaN  782.86  803.98  757.06   
3     4     NaN  790.92  809.81     NaN  837.92  769.13     NaN  761.34   
4     5     NaN  781.78  812.19     NaN  838.74  775.58     NaN  767.98   
5     6  758.49  776.99  823.28  863.99  832.84     NaN  801.46  774.82   
6     7  771.82  779.12     NaN  853.91  839.08     NaN  798.79  777.18   
7     8  773.12     NaN     NaN  842.05  836.27  764.97  793.88     NaN   
8     9  766.83     NaN  828.95  851.90     NaN  769.26  786.18     NaN   
9    10  764.69  788.80  841.06     NaN     NaN  771.96  781.74  784.28   
10   11     NaN  795.88  834.22     NaN  827.65  767.93     NaN  796.58   
11   12     NaN  792.48  835.48     NaN  826.05  784.65     NaN  793.61   
12   13  770.56  788.33  852.06  839.49  821.88     NaN  790.82  792.69   
13   14  777.66  795.12     NaN  844.57  820.38     NaN  788.22  793.32   
14   15  772.29     NaN     NaN  848.63  822.93  790.73  787.87     NaN   
15   16  772.01     NaN  836.66  859.31     NaN  796.87     NaN     NaN   
16   17  773.36  792.73  850.51  852.52     NaN  779.10  787.50  796.73   
17   18     NaN  791.95  855.09     NaN  823.86  798.53     NaN  803.40   
18   19     NaN  796.90  862.69     NaN  820.65  812.19     NaN  796.23   
19   20  771.38  797.40  867.83  849.45  819.08     NaN  787.26  784.25   
20   21  769.93  804.74     NaN  856.23     NaN     NaN  786.04  785.84   
21   22  770.50     NaN     NaN  862.06  806.17  817.33  777.66     NaN   
22   23  772.12     NaN  851.01  856.51     NaN  816.02  769.06     NaN   
23   24  771.94  809.82  861.07  858.87     NaN  817.90  767.17  790.88   
24   25     NaN  810.20  847.04     NaN  805.75  820.68     NaN  786.17   
25   26     NaN  807.96  843.96     NaN  803.74  813.25     NaN  784.44   
26   27  776.52  811.10  836.05  859.95  802.10     NaN  768.63  786.91   
27   28  787.86  816.26     NaN  856.76  816.47     NaN  768.81  784.27   
28   29  788.13     NaN     NaN  850.41  812.74     NaN  767.82     NaN   
29   30  788.18     NaN  835.23  836.62     NaN  816.36  759.18     NaN   
30   31  797.96     NaN  846.30     NaN     NaN     NaN  754.45  779.92   

       Sep     Oct     Nov     Dic  year  
0   776.46  788.15     NaN  767.29  2020  
1   770.59  785.04  771.92  760.16  2020  
2   771.45     NaN  769.17  755.34  2020  
3   773.71     NaN  758.53  752.03  2020  
4      NaN  790.81  757.16     NaN  2020  
5      NaN  794.34  752.01     NaN  2020  
6   773.76  797.35     NaN  747.61  2020  
7   773.24  795.05     NaN     NaN  2020  
8   776.65  797.25  759.25  744.82  2020  
9   767.87     NaN  753.75  739.45  2020  
10  764.99     NaN  760.90  738.17  2020  
11     NaN     NaN  757.42     NaN  2020  
12     NaN  796.05  757.43     NaN  2020  
13  768.34  797.66     NaN  733.55  2020  
14  767.01  798.56     NaN  731.58  2020  
15  760.01  801.91  766.70  734.23  2020  
16  761.54     NaN  767.86  735.09  2020  
17     NaN     NaN  767.05  723.44  2020  
18     NaN  795.68  758.10     NaN  2020  
19     NaN  788.27  758.62     NaN  2020  
20  764.34  786.66     NaN  723.85  2020  
21  772.83  784.07     NaN  730.70  2020  
22  773.40  781.41  761.55  728.96  2020  
23  781.44     NaN  765.96  716.25  2020  
24  788.02     NaN  772.83     NaN  2020  
25     NaN  777.72  771.68     NaN  2020  
26     NaN  779.57  766.00     NaN  2020  
27  787.15  772.05     NaN  710.26  2020  
28  784.19  775.56     NaN  710.64  2020  
29  784.46  770.45  766.69  711.24  2020  
30     NaN     NaN     NaN     NaN  2020

In [6]:
df2020.shape

(31, 14)

In [7]:
df2020

Día     Ene     Feb     Mar     Abr     May     Jun     Jul     Ago  \
0     1     NaN     NaN     NaN  852.03     NaN  806.32  821.23     NaN   
1     2  748.74     NaN  818.32  863.81     NaN  796.46  816.29     NaN   
2     3  754.16  799.11  817.45  861.89     NaN  782.86  803.98  757.06   
3     4     NaN  790.92  809.81     NaN  837.92  769.13     NaN  761.34   
4     5     NaN  781.78  812.19     NaN  838.74  775.58     NaN  767.98   
5     6  758.49  776.99  823.28  863.99  832.84     NaN  801.46  774.82   
6     7  771.82  779.12     NaN  853.91  839.08     NaN  798.79  777.18   
7     8  773.12     NaN     NaN  842.05  836.27  764.97  793.88     NaN   
8     9  766.83     NaN  828.95  851.90     NaN  769.26  786.18     NaN   
9    10  764.69  788.80  841.06     NaN     NaN  771.96  781.74  784.28   
10   11     NaN  795.88  834.22     NaN  827.65  767.93     NaN  796.58   
11   12     NaN  792.48  835.48     NaN  826.05  784.65     NaN  793.61   
12   13  770.56  788.33  852.06  839.49  821.88     NaN  790.82  792.69   
13   14  777.66  795.12     NaN  844.57  820.38     NaN  788.22  793.32   
14   15  772.29     NaN     NaN  848.63  822.93  790.73  787.87     NaN   
15   16  772.01     NaN  836.66  859.31     NaN  796.87     NaN     NaN   
16   17  773.36  792.73  850.51  852.52     NaN  779.10  787.50  796.73   
17   18     NaN  791.95  855.09     NaN  823.86  798.53     NaN  803.40   
18   19     NaN  796.90  862.69     NaN  820.65  812.19     NaN  796.23   
19   20  771.38  797.40  867.83  849.45  819.08     NaN  787.26  784.25   
20   21  769.93  804.74     NaN  856.23     NaN     NaN  786.04  785.84   
21   22  770.50     NaN     NaN  862.06  806.17  817.33  777.66     NaN   
22   23  772.12     NaN  851.01  856.51     NaN  816.02  769.06     NaN   
23   24  771.94  809.82  861.07  858.87     NaN  817.90  767.17  790.88   
24   25     NaN  810.20  847.04     NaN  805.75  820.68     NaN  786.17   
25   26     NaN  807.96  843.96     NaN  803.74  813.25     NaN  784.44   
26   27  776.52  811.10  836.05  859.95  802.10     NaN  768.63  786.91   
27   28  787.86  816.26     NaN  856.76  816.47     NaN  768.81  784.27   
28   29  788.13     NaN     NaN  850.41  812.74     NaN  767.82     NaN   
29   30  788.18     NaN  835.23  836.62     NaN  816.36  759.18     NaN   
30   31  797.96     NaN  846.30     NaN     NaN     NaN  754.45  779.92   

       Sep     Oct     Nov     Dic  year  
0   776.46  788.15     NaN  767.29  2020  
1   770.59  785.04  771.92  760.16  2020  
2   771.45     NaN  769.17  755.34  2020  
3   773.71     NaN  758.53  752.03  2020  
4      NaN  790.81  757.16     NaN  2020  
5      NaN  794.34  752.01     NaN  2020  
6   773.76  797.35     NaN  747.61  2020  
7   773.24  795.05     NaN     NaN  2020  
8   776.65  797.25  759.25  744.82  2020  
9   767.87     NaN  753.75  739.45  2020  
10  764.99     NaN  760.90  738.17  2020  
11     NaN     NaN  757.42     NaN  2020  
12     NaN  796.05  757.43     NaN  2020  
13  768.34  797.66     NaN  733.55  2020  
14  767.01  798.56     NaN  731.58  2020  
15  760.01  801.91  766.70  734.23  2020  
16  761.54     NaN  767.86  735.09  2020  
17     NaN     NaN  767.05  723.44  2020  
18     NaN  795.68  758.10     NaN  2020  
19     NaN  788.27  758.62     NaN  2020  
20  764.34  786.66     NaN  723.85  2020  
21  772.83  784.07     NaN  730.70  2020  
22  773.40  781.41  761.55  728.96  2020  
23  781.44     NaN  765.96  716.25  2020  
24  788.02     NaN  772.83     NaN  2020  
25     NaN  777.72  771.68     NaN  2020  
26     NaN  779.57  766.00     NaN  2020  
27  787.15  772.05     NaN  710.26  2020  
28  784.19  775.56     NaN  710.64  2020  
29  784.46  770.45  766.69  711.24  2020  
30     NaN     NaN     NaN     NaN  2020

In [8]:
month=df2020.columns[1:13]
month

Index(['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun', 'Jul', 'Ago', 'Sep', 'Oct',
       'Nov', 'Dic'],
      dtype='object')

In [9]:
df_concat=pd.concat([df2019,df2020])
df_concat

Día     Ene     Feb     Mar     Abr     May     Jun     Jul     Ago  \
0     1     NaN  657.81  651.79  678.53     NaN     NaN  679.15  700.82   
1     2  694.77     NaN     NaN  672.56  678.71     NaN  677.62  704.88   
2     3  697.09     NaN     NaN  670.05  683.31  709.80  681.30     NaN   
3     4  697.64  655.39  656.57  667.55     NaN  704.59  680.48     NaN   
4     5     NaN  653.80  659.07  666.32     NaN  696.78  677.64  711.13   
..  ...     ...     ...     ...     ...     ...     ...     ...     ...   
26   27  776.52  811.10  836.05  859.95  802.10     NaN  768.63  786.91   
27   28  787.86  816.26     NaN  856.76  816.47     NaN  768.81  784.27   
28   29  788.13     NaN     NaN  850.41  812.74     NaN  767.82     NaN   
29   30  788.18     NaN  835.23  836.62     NaN  816.36  759.18     NaN   
30   31  797.96     NaN  846.30     NaN     NaN     NaN  754.45  779.92   

       Sep     Oct     Nov     Dic  year  
0      NaN  728.21     NaN     NaN  2019  
1   720.20  729.38     NaN  812.13  2019  
2   724.72  727.80     NaN  804.29  2019  
3   725.69  719.53  735.05  802.44  2019  
4   723.30     NaN  736.66  796.75  2019  
..     ...     ...     ...     ...   ...  
26     NaN  779.57  766.00     NaN  2020  
27  787.15  772.05     NaN  710.26  2020  
28  784.19  775.56     NaN  710.64  2020  
29  784.46  770.45  766.69  711.24  2020  
30     NaN     NaN     NaN     NaN  2020  

[62 rows x 14 columns]

In [10]:
df_concat["Día"] = df_concat["Día"].astype(int)
df_concat["year"] = df_concat["year"].astype(int)
df_concat

Día     Ene     Feb     Mar     Abr     May     Jun     Jul     Ago  \
0     1     NaN  657.81  651.79  678.53     NaN     NaN  679.15  700.82   
1     2  694.77     NaN     NaN  672.56  678.71     NaN  677.62  704.88   
2     3  697.09     NaN     NaN  670.05  683.31  709.80  681.30     NaN   
3     4  697.64  655.39  656.57  667.55     NaN  704.59  680.48     NaN   
4     5     NaN  653.80  659.07  666.32     NaN  696.78  677.64  711.13   
..  ...     ...     ...     ...     ...     ...     ...     ...     ...   
26   27  776.52  811.10  836.05  859.95  802.10     NaN  768.63  786.91   
27   28  787.86  816.26     NaN  856.76  816.47     NaN  768.81  784.27   
28   29  788.13     NaN     NaN  850.41  812.74     NaN  767.82     NaN   
29   30  788.18     NaN  835.23  836.62     NaN  816.36  759.18     NaN   
30   31  797.96     NaN  846.30     NaN     NaN     NaN  754.45  779.92   

       Sep     Oct     Nov     Dic  year  
0      NaN  728.21     NaN     NaN  2019  
1   720.20  729.38     NaN  812.13  2019  
2   724.72  727.80     NaN  804.29  2019  
3   725.69  719.53  735.05  802.44  2019  
4   723.30     NaN  736.66  796.75  2019  
..     ...     ...     ...     ...   ...  
26     NaN  779.57  766.00     NaN  2020  
27  787.15  772.05     NaN  710.26  2020  
28  784.19  775.56     NaN  710.64  2020  
29  784.46  770.45  766.69  711.24  2020  
30     NaN     NaN     NaN     NaN  2020  

[62 rows x 14 columns]

In [11]:
df_sii = pd.melt(df_concat,id_vars=["year",'Día'],value_vars=month,var_name='Mes')
df_sii=df_sii[["year","Mes","Día","value"]]
df_sii.head()
df_sii.to_csv('C:/Users/franc/Documents/Magister Data science/Segundo trimestre/Analisis de datos/Tarea/tarea1-add/data/interim/df_sii.csv',
                  sep=';',index=False)



In [12]:
df_sii

year  Mes  Día   value
0    2019  Ene    1     NaN
1    2019  Ene    2  694.77
2    2019  Ene    3  697.09
3    2019  Ene    4  697.64
4    2019  Ene    5     NaN
..    ...  ...  ...     ...
739  2020  Dic   27     NaN
740  2020  Dic   28  710.26
741  2020  Dic   29  710.64
742  2020  Dic   30  711.24
743  2020  Dic   31     NaN

[744 rows x 4 columns]

In [13]:
print(df_sii.groupby(by=["Mes"],as_index = True)["value"].agg("mean"))

Mes
Abr    760.389048
Ago    749.182619
Dic    752.561750
Ene    724.854773
Feb    726.342000
Jul    735.394318
Jun    744.299024
Mar    755.523488
May    753.659500
Nov    769.540976
Oct    753.867907
Sep    748.036154
Name: value, dtype: float64


In [14]:
print(df_sii.groupby(by=["year","Mes"],as_index = False)["value"].agg("mean"))

    year  Mes       value
0   2019  Abr  667.399048
1   2019  Ago  713.703333
2   2019  Dic  770.390500
3   2019  Ene  677.061818
4   2019  Feb  656.304500
5   2019  Jul  686.059545
6   2019  Jun  692.409000
7   2019  Mar  667.678571
8   2019  May  692.003810
9   2019  Nov  776.530000
10  2019  Oct  721.032273
11  2019  Sep  718.442222
12  2020  Abr  853.379048
13  2020  Ago  784.661905
14  2020  Dic  734.733000
15  2020  Ene  772.647727
16  2020  Feb  796.379500
17  2020  Jul  784.729091
18  2020  Jun  793.718095
19  2020  Mar  839.375455
20  2020  May  821.805263
21  2020  Nov  762.884762
22  2020  Oct  788.267143
23  2020  Sep  773.402381


2- Index = False elimina el índice de la fila, el agrupar por mes, dará como resultado el valor agrupodo incluyendo los años, mientras que al dejarlo por año-mes, separa estos dos y agrupa cada mes de cada año de manera distinta.

3- Missing values :Podemos observar que la gran cantidad de días donde no hay datos del valor del dólar son durante fines de semana y esto se debe a que estos días la bolsa se encuentra cerrada por eso no hay registros de su valor.
los otros días se explica a que fueron feriados legales y por lo tanto al igual que el fin de semana estos días la bolsa de valores estuvo cerrada.

In [15]:
type(df_sii.year[0])

numpy.int32

In [16]:
def cambio_mes (param):
    if param == "Ene":
        param = "Jan"
    if param == "Feb":
        param = "Feb"
    if param == "Mar":
        param = "Mar"
    if param == "Abr":
        param = "Apr"
    if param == "May":
        param = "May"
    if param == "Jun":
        param = "Jun"
    if param == "Jul":
        param = "Jul"
    if param == "Ago":
        param = "Aug"
    if param == "Sep":
        param = "Sep"
    if param == "Oct":
        param = "Oct"
    if param == "Nov":
        param = "Nov"
    if param == "Dic":
        param = "Dec"
        
    return param

In [17]:
df_sii['Mes_en'] = df_sii["Mes"].apply(cambio_mes)
df_sii

year  Mes  Día   value Mes_en
0    2019  Ene    1     NaN    Jan
1    2019  Ene    2  694.77    Jan
2    2019  Ene    3  697.09    Jan
3    2019  Ene    4  697.64    Jan
4    2019  Ene    5     NaN    Jan
..    ...  ...  ...     ...    ...
739  2020  Dic   27     NaN    Dec
740  2020  Dic   28  710.26    Dec
741  2020  Dic   29  710.64    Dec
742  2020  Dic   30  711.24    Dec
743  2020  Dic   31     NaN    Dec

[744 rows x 5 columns]

In [18]:
df_sii['Fecha'] =  (df_sii['Día'].apply(str) +"-"+ df_sii['Mes_en'] +"-"+  df_sii['year'].apply(str))
df_sii

year  Mes  Día   value Mes_en        Fecha
0    2019  Ene    1     NaN    Jan   1-Jan-2019
1    2019  Ene    2  694.77    Jan   2-Jan-2019
2    2019  Ene    3  697.09    Jan   3-Jan-2019
3    2019  Ene    4  697.64    Jan   4-Jan-2019
4    2019  Ene    5     NaN    Jan   5-Jan-2019
..    ...  ...  ...     ...    ...          ...
739  2020  Dic   27     NaN    Dec  27-Dec-2020
740  2020  Dic   28  710.26    Dec  28-Dec-2020
741  2020  Dic   29  710.64    Dec  29-Dec-2020
742  2020  Dic   30  711.24    Dec  30-Dec-2020
743  2020  Dic   31     NaN    Dec  31-Dec-2020

[744 rows x 6 columns]

In [19]:
df_sii['Fecha'].values

array(['1-Jan-2019', '2-Jan-2019', '3-Jan-2019', '4-Jan-2019',
       '5-Jan-2019', '6-Jan-2019', '7-Jan-2019', '8-Jan-2019',
       '9-Jan-2019', '10-Jan-2019', '11-Jan-2019', '12-Jan-2019',
       '13-Jan-2019', '14-Jan-2019', '15-Jan-2019', '16-Jan-2019',
       '17-Jan-2019', '18-Jan-2019', '19-Jan-2019', '20-Jan-2019',
       '21-Jan-2019', '22-Jan-2019', '23-Jan-2019', '24-Jan-2019',
       '25-Jan-2019', '26-Jan-2019', '27-Jan-2019', '28-Jan-2019',
       '29-Jan-2019', '30-Jan-2019', '31-Jan-2019', '1-Jan-2020',
       '2-Jan-2020', '3-Jan-2020', '4-Jan-2020', '5-Jan-2020',
       '6-Jan-2020', '7-Jan-2020', '8-Jan-2020', '9-Jan-2020',
       '10-Jan-2020', '11-Jan-2020', '12-Jan-2020', '13-Jan-2020',
       '14-Jan-2020', '15-Jan-2020', '16-Jan-2020', '17-Jan-2020',
       '18-Jan-2020', '19-Jan-2020', '20-Jan-2020', '21-Jan-2020',
       '22-Jan-2020', '23-Jan-2020', '24-Jan-2020', '25-Jan-2020',
       '26-Jan-2020', '27-Jan-2020', '28-Jan-2020', '29-Jan-2020',
       '30-Ja

In [20]:
df_sii = df_sii[df_sii['value'].notnull()].loc[:, ['year','Mes', 'Día', 'value','Fecha']]
df_sii

year  Mes  Día   value        Fecha
1    2019  Ene    2  694.77   2-Jan-2019
2    2019  Ene    3  697.09   3-Jan-2019
3    2019  Ene    4  697.64   4-Jan-2019
6    2019  Ene    7  688.42   7-Jan-2019
7    2019  Ene    8  679.69   8-Jan-2019
..    ...  ...  ...     ...          ...
735  2020  Dic   23  728.96  23-Dec-2020
736  2020  Dic   24  716.25  24-Dec-2020
740  2020  Dic   28  710.26  28-Dec-2020
741  2020  Dic   29  710.64  29-Dec-2020
742  2020  Dic   30  711.24  30-Dec-2020

[499 rows x 5 columns]

In [21]:
from datetime import datetime

df_sii['Fecha'] = pd.to_datetime(df_sii['Fecha'], format='%d-%b-%Y')
df_sii


year  Mes  Día   value      Fecha
1    2019  Ene    2  694.77 2019-01-02
2    2019  Ene    3  697.09 2019-01-03
3    2019  Ene    4  697.64 2019-01-04
6    2019  Ene    7  688.42 2019-01-07
7    2019  Ene    8  679.69 2019-01-08
..    ...  ...  ...     ...        ...
735  2020  Dic   23  728.96 2020-12-23
736  2020  Dic   24  716.25 2020-12-24
740  2020  Dic   28  710.26 2020-12-28
741  2020  Dic   29  710.64 2020-12-29
742  2020  Dic   30  711.24 2020-12-30

[499 rows x 5 columns]

In [22]:
df_sii['dia_de_la_semana'] = df_sii['Fecha'].dt.day_name() 
df_sii

year  Mes  Día   value      Fecha dia_de_la_semana
1    2019  Ene    2  694.77 2019-01-02        Wednesday
2    2019  Ene    3  697.09 2019-01-03         Thursday
3    2019  Ene    4  697.64 2019-01-04           Friday
6    2019  Ene    7  688.42 2019-01-07           Monday
7    2019  Ene    8  679.69 2019-01-08          Tuesday
..    ...  ...  ...     ...        ...              ...
735  2020  Dic   23  728.96 2020-12-23        Wednesday
736  2020  Dic   24  716.25 2020-12-24         Thursday
740  2020  Dic   28  710.26 2020-12-28           Monday
741  2020  Dic   29  710.64 2020-12-29          Tuesday
742  2020  Dic   30  711.24 2020-12-30        Wednesday

[499 rows x 6 columns]

In [23]:
def dia_semana (dia):
    if dia == "Monday":
        dia = "Lunes"
    if dia == "Tuesday":
        dia = "Martes"
    if dia == "Wednesday":
        dia = "Miércoles"
    if dia == "Thursday":
        dia = "Jueves"
    if dia == "Friday":
        dia = "Viernes"
    if dia == "Saturday":
        dia = "Sábado"
    if dia == "Sunday":
        dia = "Domingo"
        
    return dia

In [24]:
df_sii['dia_de_la_semana'] = df_sii["dia_de_la_semana"].apply(dia_semana)
df_sii

year  Mes  Día   value      Fecha dia_de_la_semana
1    2019  Ene    2  694.77 2019-01-02        Miércoles
2    2019  Ene    3  697.09 2019-01-03           Jueves
3    2019  Ene    4  697.64 2019-01-04          Viernes
6    2019  Ene    7  688.42 2019-01-07            Lunes
7    2019  Ene    8  679.69 2019-01-08           Martes
..    ...  ...  ...     ...        ...              ...
735  2020  Dic   23  728.96 2020-12-23        Miércoles
736  2020  Dic   24  716.25 2020-12-24           Jueves
740  2020  Dic   28  710.26 2020-12-28            Lunes
741  2020  Dic   29  710.64 2020-12-29           Martes
742  2020  Dic   30  711.24 2020-12-30        Miércoles

[499 rows x 6 columns]

### Explora el archivo dolar_observado_bc.xlsx y su estructura.

In [48]:
df_80 = pd.read_excel('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/raw/dolar_observado_bc.xlsx',
                      skiprows=2, dtype={'Periodo':str },
                      sheet_name = '80')

In [ ]:
df_80

* Ocupa el método str.split para crear las columnas dia,mes y year en cada dataframe.

In [27]:
year_80 = df_80["Periodo"].str.split("-", expand = True)
df_80[["Fecha","Hora"]]=df_80["Periodo"].str.split(expand=True,pat=" ")

In [28]:
year_80.columns = ["Año","Mes","Día"]

year_80["Día"] = year_80["Día"].str.slice(0, 2)

df_80 = pd.concat([df_80,year_80],axis=1)

df_80

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0     1982-08-09 00:00:00              55.65  1982-08-09  00:00:00  1982  08   
1     1982-08-10 00:00:00              62.79  1982-08-10  00:00:00  1982  08   
2     1982-08-11 00:00:00              64.07  1982-08-11  00:00:00  1982  08   
3     1982-08-12 00:00:00              60.52  1982-08-12  00:00:00  1982  08   
4     1982-08-13 00:00:00              56.87  1982-08-13  00:00:00  1982  08   
...                   ...                ...         ...       ...   ...  ..   
2697  1989-12-27 00:00:00             296.62  1989-12-27  00:00:00  1989  12   
2698  1989-12-28 00:00:00             296.69  1989-12-28  00:00:00  1989  12   
2699  1989-12-29 00:00:00             296.58  1989-12-29  00:00:00  1989  12   
2700  1989-12-30 00:00:00                NaN  1989-12-30  00:00:00  1989  12   
2701  1989-12-31 00:00:00                NaN  1989-12-31  00:00:00  1989  12   

     Día  
0     09  
1     10  
2     11  
3     12  
4     13  
...   ..  
2697  27  
2698  28  
2699  29  
2700  30  
2701  31  

[2702 rows x 7 columns]

#### Decada 90

In [29]:
df_90 = pd.read_excel('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/raw/dolar_observado_bc.xlsx',
                      skiprows=2, dtype={'Periodo':str },
                      sheet_name = '90')

In [30]:
year_90 = df_90["Periodo"].str.split("-", expand = True)
df_90[["Fecha","Hora"]]=df_90["Periodo"].str.split(expand=True,pat=" ")

year_90.columns = ["Año","Mes","Día"]

year_90["Día"] = year_90["Día"].str.slice(0, 2)

df_90 = pd.concat([df_90,year_90],axis=1)

df_90

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0     1990-01-01 00:00:00                NaN  1990-01-01  00:00:00  1990  01   
1     1990-01-02 00:00:00             297.37  1990-01-02  00:00:00  1990  01   
2     1990-01-03 00:00:00             297.32  1990-01-03  00:00:00  1990  01   
3     1990-01-04 00:00:00             297.65  1990-01-04  00:00:00  1990  01   
4     1990-01-05 00:00:00             297.66  1990-01-05  00:00:00  1990  01   
...                   ...                ...         ...       ...   ...  ..   
3647  1999-12-27 00:00:00             534.19  1999-12-27  00:00:00  1999  12   
3648  1999-12-28 00:00:00             531.22  1999-12-28  00:00:00  1999  12   
3649  1999-12-29 00:00:00             530.44  1999-12-29  00:00:00  1999  12   
3650  1999-12-30 00:00:00             527.70  1999-12-30  00:00:00  1999  12   
3651  1999-12-31 00:00:00                NaN  1999-12-31  00:00:00  1999  12   

     Día  
0     01  
1     02  
2     03  
3     04  
4     05  
...   ..  
3647  27  
3648  28  
3649  29  
3650  30  
3651  31  

[3652 rows x 7 columns]

#### Decada 00

In [31]:
df_00 = pd.read_excel('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/raw/dolar_observado_bc.xlsx',
                      skiprows=2, dtype={'Periodo':str },
                      sheet_name = '00')

year_00 = df_00["Periodo"].str.split("-", expand = True)
df_00[["Fecha","Hora"]]=df_00["Periodo"].str.split(expand=True,pat=" ")
year_00.columns = ["Año","Mes","Día"]

year_00["Día"] = year_00["Día"].str.slice(0, 2)

df_00 = pd.concat([df_00,year_00],axis=1)

df_00

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0     2000-01-01 00:00:00                NaN  2000-01-01  00:00:00  2000  01   
1     2000-01-02 00:00:00                NaN  2000-01-02  00:00:00  2000  01   
2     2000-01-03 00:00:00             530.07  2000-01-03  00:00:00  2000  01   
3     2000-01-04 00:00:00             527.28  2000-01-04  00:00:00  2000  01   
4     2000-01-05 00:00:00             529.08  2000-01-05  00:00:00  2000  01   
...                   ...                ...         ...       ...   ...  ..   
3648  2009-12-27 00:00:00                NaN  2009-12-27  00:00:00  2009  12   
3649  2009-12-28 00:00:00             506.62  2009-12-28  00:00:00  2009  12   
3650  2009-12-29 00:00:00             506.27  2009-12-29  00:00:00  2009  12   
3651  2009-12-30 00:00:00             506.43  2009-12-30  00:00:00  2009  12   
3652  2009-12-31 00:00:00                NaN  2009-12-31  00:00:00  2009  12   

     Día  
0     01  
1     02  
2     03  
3     04  
4     05  
...   ..  
3648  27  
3649  28  
3650  29  
3651  30  
3652  31  

[3653 rows x 7 columns]

#### Decada 10

In [32]:
df_10 = pd.read_excel('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/raw/dolar_observado_bc.xlsx',
                      skiprows=2, dtype={'Periodo':str },
                      sheet_name = '10')

year_10 = df_10["Periodo"].str.split("-", expand = True)
df_10[["Fecha","Hora"]]=df_10["Periodo"].str.split(expand=True,pat=" ")
year_10.columns = ["Año","Mes","Día"]

year_10["Día"] = year_10["Día"].str.slice(0, 2)

df_10 = pd.concat([df_10,year_10],axis=1)

df_10

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0     2010-01-01 00:00:00                NaN  2010-01-01  00:00:00  2010  01   
1     2010-01-02 00:00:00                NaN  2010-01-02  00:00:00  2010  01   
2     2010-01-03 00:00:00                NaN  2010-01-03  00:00:00  2010  01   
3     2010-01-04 00:00:00             507.10  2010-01-04  00:00:00  2010  01   
4     2010-01-05 00:00:00             505.70  2010-01-05  00:00:00  2010  01   
...                   ...                ...         ...       ...   ...  ..   
3647  2019-12-27 00:00:00             746.66  2019-12-27  00:00:00  2019  12   
3648  2019-12-28 00:00:00                NaN  2019-12-28  00:00:00  2019  12   
3649  2019-12-29 00:00:00                NaN  2019-12-29  00:00:00  2019  12   
3650  2019-12-30 00:00:00             744.62  2019-12-30  00:00:00  2019  12   
3651  2019-12-31 00:00:00                NaN  2019-12-31  00:00:00  2019  12   

     Día  
0     01  
1     02  
2     03  
3     04  
4     05  
...   ..  
3647  27  
3648  28  
3649  29  
3650  30  
3651  31  

[3652 rows x 7 columns]

#### Decada 20

In [33]:
df_20 = pd.read_excel('https://raw.githubusercontent.com/Franciscosanchezl/tarea1-add/master/data/raw/dolar_observado_bc.xlsx',
                      skiprows=2, dtype={'Periodo':str },
                      sheet_name = '20')

year_20 = df_20["Periodo"].str.split("-", expand = True)

year_20.columns = ["Año","Mes","Día"]
df_20[["Fecha","Hora"]]=df_20["Periodo"].str.split(expand=True,pat=" ")
year_20["Día"] = year_20["Día"].str.slice(0, 2)

df_20 = pd.concat([df_20,year_20],axis=1)

df_20

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0    2020-01-01 00:00:00                NaN  2020-01-01  00:00:00  2020  01   
1    2020-01-02 00:00:00             748.74  2020-01-02  00:00:00  2020  01   
2    2020-01-03 00:00:00             754.16  2020-01-03  00:00:00  2020  01   
3    2020-01-04 00:00:00                NaN  2020-01-04  00:00:00  2020  01   
4    2020-01-05 00:00:00                NaN  2020-01-05  00:00:00  2020  01   
..                   ...                ...         ...       ...   ...  ..   
969  2022-08-27 00:00:00                NaN  2022-08-27  00:00:00  2022  08   
970  2022-08-28 00:00:00                NaN  2022-08-28  00:00:00  2022  08   
971  2022-08-29 00:00:00             900.22  2022-08-29  00:00:00  2022  08   
972  2022-08-30 00:00:00             890.90  2022-08-30  00:00:00  2022  08   
973  2022-08-31 00:00:00             882.11  2022-08-31  00:00:00  2022  08   

    Día  
0    01  
1    02  
2    03  
3    04  
4    05  
..   ..  
969  27  
970  28  
971  29  
972  30  
973  31  

[974 rows x 7 columns]

* Crea un dataframe que una todos los dataframes en uno solo con las mismas columnas, y nombralo df_bc_all.

In [34]:
df_bc_all = pd.concat([df_80,df_90,df_00,df_10,df_20], axis=0)
df_bc_all = df_bc_all.reset_index()
df_bc_all = df_bc_all.iloc[: , 1:]
df_bc_all

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
0      1982-08-09 00:00:00              55.65  1982-08-09  00:00:00  1982  08   
1      1982-08-10 00:00:00              62.79  1982-08-10  00:00:00  1982  08   
2      1982-08-11 00:00:00              64.07  1982-08-11  00:00:00  1982  08   
3      1982-08-12 00:00:00              60.52  1982-08-12  00:00:00  1982  08   
4      1982-08-13 00:00:00              56.87  1982-08-13  00:00:00  1982  08   
...                    ...                ...         ...       ...   ...  ..   
14628  2022-08-27 00:00:00                NaN  2022-08-27  00:00:00  2022  08   
14629  2022-08-28 00:00:00                NaN  2022-08-28  00:00:00  2022  08   
14630  2022-08-29 00:00:00             900.22  2022-08-29  00:00:00  2022  08   
14631  2022-08-30 00:00:00             890.90  2022-08-30  00:00:00  2022  08   
14632  2022-08-31 00:00:00             882.11  2022-08-31  00:00:00  2022  08   

      Día  
0      09  
1      10  
2      11  
3      12  
4      13  
...    ..  
14628  27  
14629  28  
14630  29  
14631  30  
14632  31  

[14633 rows x 7 columns]

* Crea una carpeta dentro de la carpeta data/processed llamada dolar_bc_year, agrupa por year usando groupby y guarda un CSV para cada año desde 1982 hasta 2022, con los nombres dolar_bc_YYYY.csv donde YYYY es el año. Busca una manera de hacerlo iterando y no uno a uno.

In [35]:
df_año=pd.DataFrame()
años=sorted(set(df_bc_all['Año']))
for año in años:
    df_año =  df_bc_all.loc[df_bc_all['Año'] == año]
    df_año.to_csv(f'C:/Users/franc/Documents/Magister Data science/Segundo trimestre/Analisis de datos/Tarea/tarea1-add/data/processed/dolar_bc_year/dolar_bc_{año}.csv')
    

In [36]:
df_bc_2019_2020= df_bc_all[(df_bc_all['Año'] == '2019') | (df_bc_all['Año'] == '2020')]
df_bc_2019_2020

Periodo  1.Dólar observado       Fecha      Hora   Año Mes  \
13294  2019-01-01 00:00:00                NaN  2019-01-01  00:00:00  2019  01   
13295  2019-01-02 00:00:00             694.77  2019-01-02  00:00:00  2019  01   
13296  2019-01-03 00:00:00             697.09  2019-01-03  00:00:00  2019  01   
13297  2019-01-04 00:00:00             697.64  2019-01-04  00:00:00  2019  01   
13298  2019-01-05 00:00:00                NaN  2019-01-05  00:00:00  2019  01   
...                    ...                ...         ...       ...   ...  ..   
14020  2020-12-27 00:00:00                NaN  2020-12-27  00:00:00  2020  12   
14021  2020-12-28 00:00:00             710.26  2020-12-28  00:00:00  2020  12   
14022  2020-12-29 00:00:00             710.64  2020-12-29  00:00:00  2020  12   
14023  2020-12-30 00:00:00             711.24  2020-12-30  00:00:00  2020  12   
14024  2020-12-31 00:00:00                NaN  2020-12-31  00:00:00  2020  12   

      Día  
13294  01  
13295  02  
13296  03  
13297  04  
13298  05  
...    ..  
14020  27  
14021  28  
14022  29  
14023  30  
14024  31  

[731 rows x 7 columns]

## los datafreme son diferentes, ya que, el **df_bc_2019_2020** contiene los anios agrupados por meses y el valor promedio de cada mes,  En cambio el dataframe **df_sii** contiene 499 filas × 6 columnas dando los valores diarios y con otra informacion.

 faltaaaaa  Haz un merge de la columna dia_de_la_semana del df_sii al dataframe df_bc_all. Luego, extiende la variable dia_de_la_semana para todos los valores. Finalmente, crea una variable con las combinaciones [dia,mes] que tienen al menos un NaN que sea en día de semana (es decir, excluyendo Sábado y Domingo). Luego, cuenta cuantos NaN(incluyendo Sábados y Domingos) hay para cada combinación[dia/mes]. ¿Qué puedes inferir de los días con mayor cantidad de NaN?

In [37]:
df_bc_all["Fecha"]=pd.to_datetime(df_bc_all["Fecha"])
df_bc_all.head()

Periodo  1.Dólar observado      Fecha      Hora   Año Mes Día
0  1982-08-09 00:00:00              55.65 1982-08-09  00:00:00  1982  08  09
1  1982-08-10 00:00:00              62.79 1982-08-10  00:00:00  1982  08  10
2  1982-08-11 00:00:00              64.07 1982-08-11  00:00:00  1982  08  11
3  1982-08-12 00:00:00              60.52 1982-08-12  00:00:00  1982  08  12
4  1982-08-13 00:00:00              56.87 1982-08-13  00:00:00  1982  08  13

In [38]:
df_nueva=df_bc_all.merge(df_sii,left_on="Fecha",right_on="Fecha",how="left")
df_nueva.describe()
df_nueva

Periodo  1.Dólar observado      Fecha      Hora   Año  \
0      1982-08-09 00:00:00              55.65 1982-08-09  00:00:00  1982   
1      1982-08-10 00:00:00              62.79 1982-08-10  00:00:00  1982   
2      1982-08-11 00:00:00              64.07 1982-08-11  00:00:00  1982   
3      1982-08-12 00:00:00              60.52 1982-08-12  00:00:00  1982   
4      1982-08-13 00:00:00              56.87 1982-08-13  00:00:00  1982   
...                    ...                ...        ...       ...   ...   
14628  2022-08-27 00:00:00                NaN 2022-08-27  00:00:00  2022   
14629  2022-08-28 00:00:00                NaN 2022-08-28  00:00:00  2022   
14630  2022-08-29 00:00:00             900.22 2022-08-29  00:00:00  2022   
14631  2022-08-30 00:00:00             890.90 2022-08-30  00:00:00  2022   
14632  2022-08-31 00:00:00             882.11 2022-08-31  00:00:00  2022   

      Mes_x Día_x  year Mes_y  Día_y  value dia_de_la_semana  
0        08    09   NaN   NaN    NaN    NaN              NaN  
1        08    10   NaN   NaN    NaN    NaN              NaN  
2        08    11   NaN   NaN    NaN    NaN              NaN  
3        08    12   NaN   NaN    NaN    NaN              NaN  
4        08    13   NaN   NaN    NaN    NaN              NaN  
...     ...   ...   ...   ...    ...    ...              ...  
14628    08    27   NaN   NaN    NaN    NaN              NaN  
14629    08    28   NaN   NaN    NaN    NaN              NaN  
14630    08    29   NaN   NaN    NaN    NaN              NaN  
14631    08    30   NaN   NaN    NaN    NaN              NaN  
14632    08    31   NaN   NaN    NaN    NaN              NaN  

[14633 rows x 12 columns]

In [39]:
df_nueva["dia_de_la_semana"].fillna(df_nueva['Fecha'].dt.day_name() , inplace = True) 

In [40]:
def dia_semana (dia):
    if dia == "Monday":
        dia = "Lunes"
    if dia == "Tuesday":
        dia = "Martes"
    if dia == "Wednesday":
        dia = "Miércoles"
    if dia == "Thursday":
        dia = "Jueves"
    if dia == "Friday":
        dia = "Viernes"
    if dia == "Saturday":
        dia = "Sábado"
    if dia == "Sunday":
        dia = "Domingo"
    else:
        dia=dia
        
    return dia
df_nueva['dia_de_la_semana'] = df_nueva["dia_de_la_semana"].apply(dia_semana)


In [41]:
df_nueva_2=df_nueva.loc[df_nueva["dia_de_la_semana"]!="Domingo"]
df_nueva_2=df_nueva_2.loc[df_nueva["dia_de_la_semana"]!="Sábado"]
df_nueva_2["dia_de_la_semana"].unique()

array(['Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes'], dtype=object)

In [42]:
df_nueva_2["Mes-Dia"]=df_nueva_2["Mes_x"].apply(str)+"-"+df_nueva_2["Día_x"].apply(str)


In [43]:
df2 = df_nueva_2["1.Dólar observado"].isnull().groupby([df_nueva_2["Mes-Dia"]]).sum().astype(int).reset_index(name='count')
df2.loc[df2["count"]>0].sort_values("count",ascending=False)

Mes-Dia  count
305   11-01     29
227   08-15     29
0     01-01     28
121   05-01     28
359   12-25     28
..      ...    ...
1     01-02      1
163   06-12      1
165   06-14      1
166   06-15      1
161   06-10      1

[78 rows x 2 columns]

Me sorprende que el máximo de valores nulos es 30 y no 40, pues el rango de fechas va desde el 82 hasta el 2022.
Por otro lado probablemente son fechas feriadas, como podemos analizar al ver el 1 de noviembre.

3.6

Calcula el promedio del precio del dolar en los días de cumpleaños del grupo para cada año entre 1983 y 2021 y el promedio del precio del dolar para cada año completo entre 1983 y 2021, y muestra una tabla donde se vean ambos resultados. ¿Qué diferencias hay en ambos indicadores en cada año?

In [44]:
# Fechas de cumpleanios
fecha1_mes = ['04']
fecha1_dia = ['18']

fecha2_mes = ['06']
fecha2_dia = ['23']

fecha3_mes = ['03']
fecha3_dia = ['22']

fecha4_mes = ['01']
fecha4_dia = ['10']

fecha5_mes = ['01']
fecha5_dia = ['23']

df_cumpleanios = df_bc_all[ (df_bc_all.Mes.isin(fecha1_mes) & df_bc_all["Día"].isin(fecha1_dia)) |
                            (df_bc_all.Mes.isin(fecha2_mes) & df_bc_all["Día"].isin(fecha2_dia)) |
                            (df_bc_all.Mes.isin(fecha3_mes) & df_bc_all["Día"].isin(fecha3_dia)) |
                            (df_bc_all.Mes.isin(fecha4_mes) & df_bc_all["Día"].isin(fecha4_dia)) |
                            (df_bc_all.Mes.isin(fecha5_mes) & df_bc_all["Día"].isin(fecha5_dia))
                          ]

df_cumpleanios = df_cumpleanios[(df_cumpleanios['Año'] != '1982') & (df_cumpleanios['Año'] != '2022')]
df_cumpleanios

Periodo  1.Dólar observado      Fecha      Hora   Año Mes  \
154    1983-01-10 00:00:00              73.90 1983-01-10  00:00:00  1983  01   
167    1983-01-23 00:00:00                NaN 1983-01-23  00:00:00  1983  01   
225    1983-03-22 00:00:00              73.54 1983-03-22  00:00:00  1983  03   
252    1983-04-18 00:00:00              73.61 1983-04-18  00:00:00  1983  04   
318    1983-06-23 00:00:00              77.37 1983-06-23  00:00:00  1983  06   
...                    ...                ...        ...       ...   ...  ..   
14034  2021-01-10 00:00:00                NaN 2021-01-10  00:00:00  2021  01   
14047  2021-01-23 00:00:00                NaN 2021-01-23  00:00:00  2021  01   
14105  2021-03-22 00:00:00             717.96 2021-03-22  00:00:00  2021  03   
14132  2021-04-18 00:00:00                NaN 2021-04-18  00:00:00  2021  04   
14198  2021-06-23 00:00:00             749.34 2021-06-23  00:00:00  2021  06   

      Día  
154    10  
167    23  
225    22  
252    18  
318    23  
...    ..  
14034  10  
14047  23  
14105  22  
14132  18  
14198  23  

[195 rows x 7 columns]

In [45]:
df_cumpleanios_grupo = df_cumpleanios.groupby(by=["Año"]  ,as_index = False)["1.Dólar observado"].mean()
df_cumpleanios_grupo

Año  1.Dólar observado
0   1983          74.605000
1   1984          88.147500
2   1985         138.442500
3   1986         186.687500
4   1987         213.165000
5   1988         244.350000
6   1989         253.686000
7   1990         297.605000
8   1991         338.880000
9   1992         369.270000
10  1993         400.605000
11  1994         425.292500
12  1995         397.906000
13  1996         409.520000
14  1997         420.550000
15  1998         456.575000
16  1999         500.335000
17  2000         519.095000
18  2001         582.492500
19  2002         660.137500
20  2003         716.683333
21  2004         605.380000
22  2005         578.132500
23  2006         528.984000
24  2007         536.855000
25  2008         479.717500
26  2009         578.360000
27  2010         530.315000
28  2011         480.502500
29  2012         493.700000
30  2013         472.357500
31  2014         545.116667
32  2015         628.295000
33  2016         672.306667
34  2017         659.828000
35  2018         604.145000
36  2019         668.987500
37  2020         784.276667
38  2021         733.650000

In [46]:
df_bc_all_1983_2021 =  df_bc_all[(df_bc_all['Año'] != '1982') & (df_bc_all['Año'] != '2022')]
df_bc_all_1983_2021 = df_bc_all_1983_2021.groupby(by=["Año"]  ,as_index = False)["1.Dólar observado"].mean()

In [47]:
columns_comparate = ['anio_cumpleanios', 'mean_cumpleanios','anio_df_bc_all','mean_df_bc_all']
df_comparate = pd.concat([df_cumpleanios_grupo,df_bc_all_1983_2021], axis = 1)
df_comparate.columns = columns_comparate
df_comparate

anio_cumpleanios  mean_cumpleanios anio_df_bc_all  mean_df_bc_all
0              1983         74.605000           1983       78.766561
1              1984         88.147500           1984       98.233494
2              1985        138.442500           1985      160.725840
3              1986        186.687500           1986      192.868112
4              1987        213.165000           1987      219.456129
5              1988        244.350000           1988      244.987968
6              1989        253.686000           1989      266.494777
7              1990        297.605000           1990      304.679757
8              1991        338.880000           1991      349.114859
9              1992        369.270000           1992      362.696145
10             1993        400.605000           1993      404.255238
11             1994        425.292500           1994      420.252421
12             1995        397.906000           1995      396.782056
13             1996        409.520000           1996      412.219040
14             1997        420.550000           1997      419.248795
15             1998        456.575000           1998      460.319719
16             1999        500.335000           1999      508.897251
17             2000        519.095000           2000      538.871129
18             2001        582.492500           2001      634.428502
19             2002        660.137500           2002      689.242450
20             2003        716.683333           2003      691.535560
21             2004        605.380000           2004      609.549960
22             2005        578.132500           2005      559.862778
23             2006        528.984000           2006      530.263052
24             2007        536.855000           2007      522.690688
25             2008        479.717500           2008      521.789560
26             2009        578.360000           2009      559.667320
27             2010        530.315000           2010      510.376640
28             2011        480.502500           2011      483.364048
29             2012        493.700000           2012      486.746559
30             2013        472.357500           2013      494.995161
31             2014        545.116667           2014      570.005904
32             2015        628.295000           2015      654.249000
33             2016        672.306667           2016      676.832421
34             2017        659.828000           2017      649.328785
35             2018        604.145000           2018      640.290772
36             2019        668.987500           2019      702.631048
37             2020        784.276667           2020      792.221833
38             2021        733.650000           2021      759.272840

Las diferencias que se encontraron en cada uno de los DataFrame son minimas, La observacion es relevante, dado que se redujo de manera importante la cantidad de datos en la que se calculo el promedio para cada uno de los datos.